tests/integration/timeout_tests.py

```diff
@@ -23,18 +27,39 @@ def test_default_timeout():
 
 
 def test_route_specific_timeout():
-    baseplate = Baseplate()
+    baseplate = _create_baseplate_object("5 milliseconds")
+    context = baseplate.make_context_object()
+    with baseplate.make_server_span(context, "test"):
+        with pytest.raises(ServerTimeout):
+            gevent.sleep(1)
 
-    observer = TimeoutBaseplateObserver.from_config(
-        {"server_timeout.default": "1 hour", "server_timeout.by_endpoint.test": "5 milliseconds"}
-    )
-    baseplate.register(observer)
+    context = baseplate.make_context_object()
+    with baseplate.make_server_span(context, "test"):
+        gevent.sleep(0)  #
 
+
+def test_timeout_from_context():
+    baseplate = _create_baseplate_object("1 hour")
+    baseplate.add_to_context("deadline_budget", 0.01)
+
+    # tests short deadline_budget causes timeout
     context = baseplate.make_context_object()
     with baseplate.make_server_span(context, "test"):
         with pytest.raises(ServerTimeout):
             gevent.sleep(1)
 
+    # tests long deadline_budget and server timeout doesn't timeout
     context = baseplate.make_context_object()
+    baseplate.add_to_context("deadline_budget", 999999999999)
     with baseplate.make_server_span(context, "test"):
-        gevent.sleep(0)  # shouldn't time out since it's so fast!
+        gevent.sleep(1)  # shouldn't time out since timeouts are so long
```

Comment:
This means this test would take at least 1 second to run, which is in general not a good idea. Try avoid long sleeps in tests, use something like `gevent.sleep(0.01)` and a deadline_budget higher than 10 millisecond (for example, "1000") instead.

(I think the other sleeps with a shorter timeout would be better because they will timeout before sleep for 1 full second.)